starkcore/utils/rest/rest.go

```diff
@@ -0,0 +1,323 @@
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+	Error "github.com/starkinfra/core-go/starkcore/error"
+	"github.com/starkinfra/core-go/starkcore/user/user"
+	"github.com/starkinfra/core-go/starkcore/utils/api"
+	"github.com/starkinfra/core-go/starkcore/utils/request"
+	"io/ioutil"
+	"math"
+	"strconv"
+)
+
+func GetPage(sdkVersion string, host string, apiVersion string, language string, timeout int, user user.User, resource map[string]string, query map[string]interface{}) ([]byte, string, Error.StarkErrors) {
+	data := map[string]interface{}{}
+	page, err := request.Fetch(
+		host,
+		sdkVersion,
+		user,
+		"GET",
+		api.Endpoint(resource),
+		"",
+		apiVersion,
+		language,
+		timeout,
+		query,
+	)
+	if err.Errors != nil {
+		return nil, "", err
+	}
+	resp, _ := ioutil.ReadAll(page.Body)
+	unmarshalError := json.Unmarshal(resp, &data)
+	if unmarshalError != nil {
+		panic(unmarshalError)
+	}
+	cursor := data["cursor"]
+	jsonBytes, _ := json.Marshal(data[api.LastNamePlural(resource)])
+	if cursor == nil {
+		return jsonBytes, "", err
+	}
+	return jsonBytes, cursor.(string), err
+}
+
+func GetStream(sdkVersion string, host string, apiVersion string, language string, timeout int, user user.User, resource map[string]string, query map[string]interface{}, c chan map[string]interface{}, e chan Error.StarkError) {
+	var response []map[string]interface{}
+	datas := make([]map[string]interface{}, len(response))
+	var isNilCursor, isValid bool
+	limitQuery := make(map[string]interface{})
+	limit, _ := strconv.Atoi(fmt.Sprintf("%v", query["limit"]))
+	if limit == 0 {
+		isValid = true
+		limitQuery["limit"] = nil
+	}
+	for k, v := range query {
+		limitQuery[k] = v
+	}
+	go func() {
+		defer close(c)
+		if limit != 0 {
+			limitQuery["limit"] = int(math.Min(float64(limit), 100))
+		}
+		for _ = 0; (limit > 0 && isNilCursor == false) || isValid == true && limitQuery["limit"] == nil; {
```

Comment:
Não da pra fazer algo tipo:
```suggestion
		for _ = 0; (limit > 0 && !isNilCursor) || isValid && limitQuery["limit"] == nil; {
```
?

Reply:
Bom botar um parênteses pra ficar mais claro qual a ordem de processamento no final:
seria `|| (isValid && limitQuery["limit"] == nil)`?

Reply:
Sim, ficou bem mais clean mesmo